examples/dynamic_rendering/source/dynamic_rendering.cpp

```diff
@@ -0,0 +1,244 @@
+				avk::context().main_window()->backbuffer_reference_at_index(0)) // Get scissor and viewport settings from current window backbuffer
+				.enable_dynamic_viewport(),                                     // But also enable dynamic viewport
+			avk::cfg::dynamic_rendering::enabled,
+			colorAttachmentDescription
+		);
+
+		// Get hold of the "ImGui Manager" and add a callback that draws UI elements:
+		auto imguiManager = avk::current_composition()->element_by_type<avk::imgui_manager>();
+		if (nullptr != imguiManager) {
+			imguiManager->add_callback([this, r](){
+		        ImGui::Begin("Info & Settings");
+				ImGui::SetWindowPos(ImVec2(1.0f, 1.0f), ImGuiCond_FirstUseEver);
+				ImGui::Text("%.3f ms/frame", 1000.0f / ImGui::GetIO().Framerate);
+				ImGui::Text("%.1f FPS", ImGui::GetIO().Framerate);
+				for(uint32_t i = 0; i < mClearColors.size(); i++)
+				{
+					ImGui::ColorEdit4((std::string("Clear color renderpass ") + std::to_string(i)).c_str(), mClearColors.at(i).data());
+				}
+				const float minXSize = (1.0f + mBorderThickness) / static_cast<float>(r.x);
+				const float minYSize = (1.0f + mBorderThickness) / static_cast<float>(r.y);
+				ImGui::SliderFloat("Renderpass x split", &mXSplit, minXSize, 1.0f - minXSize, "%.3f", ImGuiSliderFlags_AlwaysClamp);
+				ImGui::SliderFloat("Renderpass y split", &mYSplit, minYSize, 1.0f - minYSize, "%.3f", ImGuiSliderFlags_AlwaysClamp);
+				ImGui::SliderInt("Border Thickness", reinterpret_cast<int32_t*>(&mBorderThickness), 0, 10);
+				ImGui::Checkbox("Fullscreen viewport", &mFullscreenViewport);
+				ImGui::End();
+			});
+		}
+	}
+
+	void update() override
+	{
+		// On Esc pressed,
+		if (avk::input().key_pressed(avk::key_code::escape) || avk::context().main_window()->should_be_closed()) {
+			// stop the current composition:
+			avk::current_composition()->stop();
+		}
+	}
+
+	void render() override
+	{
+		auto mainWnd = avk::context().main_window();
+		auto inFlightIndex = mainWnd->in_flight_index_for_frame();
+
+		// Get a command pool to allocate command buffers from:
+		auto& commandPool = avk::context().get_command_pool_for_single_use_command_buffers(*mQueue);
+
+		// The swap chain provides us with an "image available semaphore" for the current frame.
+		// Only after the swapchain image has become available, we may start rendering into it.
+		auto imageAvailableSemaphore = mainWnd->consume_current_image_available_semaphore();
+		
+		// Create a single command buffer:
+		auto cmdBfr = commandPool->alloc_command_buffers(1u, vk::CommandBufferUsageFlagBits::eOneTimeSubmit);
+		
+		avk::image_view swapchainImageView = mainWnd->current_image_view();
+		avk::image_t const & swapchainImage = mainWnd->current_image_reference(); 
+		avk::attachment colorAttachment = avk::attachment::declare_for(swapchainImageView.as_reference(), avk::on_load::clear, avk::usage::color(0), avk::on_store::store);
+
+		std::vector<avk::recorded_commands_t> render_commands = {};
```

Comment:
```suggestion
		std::vector<avk::recorded_commands_t> renderCommands = {};
```

lowerCamelCase plz :)